src/provider/provider_coarse.c

```diff
@@ -477,16 +756,28 @@ static enum umf_result_t coarse_memory_provider_alloc(void *provider,
                        coarse_provider->alloc_size);
             }
 
+            rv = allocated_blocks_add(coarse_provider->allocated_blocks, curr);
```

Comment:
move these calls above the printf so it will be printed only when this function succeeds - maybe closer to the "block_list_add"

Reply:
Done